tests/gif/bug00227.c

```diff
@@ -60,9 +60,20 @@ int main()
 	fp = fopen(file, "rb");
 	if (!fp) return 1;
 	buf = malloc(PROBE_SIZE);
-	if (!buf) return 1;
-	if (PROBE_SIZE != fread(buf, 1, PROBE_SIZE, fp)) return 1;
-	if (buf[PROBE_SIZE-1] & 0x80) return 1;
+	if (!buf) {
+		fclose(fp);		
+		return 1;
+	}
+	if (PROBE_SIZE != fread(buf, 1, PROBE_SIZE, fp)){ 
+		free(buf);
+		fclose(fp);		
+		return 1;
+	}
+	if (buf[PROBE_SIZE-1] & 0x80){ 
+		free(buf);
+		fclose(fp);
+		return 1
+	};
```

Comment:
I suggest to use `goto` to make code simple. `goto` is not recommanded in lib code, because it will make the code hard to maintain. But considering the low complexity of the test case code, I think it is fine.

Reply:
And please take attention to the CI failure. :)